test/test_utils/__init__.py

```diff
@@ -1205,21 +1205,14 @@ def get_cuda_version_from_tag(image_uri):
     :return: cuda version as cuXXX
     """
     cuda_framework_version = None
-
     cuda_str = ["cu", "gpu"]
-    if all(keyword in image_uri for keyword in cuda_str):
-        cuda_framework_version = re.search(r"(cu\d+)-", image_uri).groups()[0]
-
-    if not cuda_framework_version and "gpu" in image_uri:
-        response_output = json.loads(run(f"docker inspect {image_uri}", hide=True).stdout)
-        env_vars = response_output[0]['ContainerConfig']['Env']
-        cuda_version_env_variables = [env_var for env_var in env_vars if env_var.startswith("CUDA_VERSION=")]
-        if not cuda_version_env_variables:
-            raise RuntimeError(f"Image {image_uri} does not have an Env variable called CUDA_VERSION")
-        cuda_version_string = cuda_version_env_variables[0]
-        cuda_version = cuda_version_string.split("=")[1]
-        image_cuda_version = ''.join(cuda_version.split('.')[:2])
-        cuda_framework_version = f"cu{image_cuda_version}"
+    response_output = json.loads(run(f"docker inspect {image_uri}", hide=True).stdout)
+    repo_tags = response_output[0]["RepoTags"]
+
+    for repo_tag in repo_tags:
+        if all(keyword in repo_tag for keyword in cuda_str):
+            cuda_framework_version = re.search(r"(cu\d+)-", repo_tag).groups()[0]
+            return cuda_framework_version
```

Comment:
1. Please use `ecr.describe_images` rather than assuming that the image is downloaded.
2. Please add a `raise CudaVersionTagNotFoundException` or something if the for loop completes for a GPU image, and return `None` if the image is not a GPU image type.